File: Data Structures & Algorithms/partition-equal-subset-sum/submission-2.java
class Solution {
    public boolean canPartition(int[] nums) {
        int s =0;

        for(int x : nums) s+=x;
        if(s%2!=0) return false;
        s=s/2;
        boolean[] dp = new boolean[s+1];
        dp[0] =true;
        for(int t : nums){
        for(int i=s;i>=0;i--){
                if(i-t>=0) dp[i] = dp[i] || dp[i-t];
            }
        }

        return dp[s];
    }
}

File: Data Structures & Algorithms/majority-element/submission-2.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans = nums[0];
        int count=1;
        for(int i=1;i<nums.size();i++){
            int x = nums[i];
            if(x == ans){
                count++;
            }else{
                count--;
                if(count<=0) ans = x;
            }
            cout<<count<<endl;
        }

        return ans;
        
    }
};